### PR TITLE
AssignmentPattern

### PR DESCRIPTION
**Pull request by hzoo**
Fixes https://github.com/cst/cst/issues/41

Can probably add more tests.

Is there a site/docs/something that has coverage over all syntax we could use? Then we can get more coverage.

I see https://github.com/estree/estree/issues/66 for adding examples

### CODE REVIEW

File: test/lib/elements/types/AssignmentPattern.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        let params = parseAndGetStatementInFunctionParams(' x = 1 + 2 ');
+        expect(params[0].left.type).to.equal('Identifier');
+        expect(params[0].left.name).to.equal('x');
+        expect(params[0].right.type).to.equal('BinaryExpression');
+        expect(params[0].right.left.type).to.equal('Literal');
+        expect(params[0].right.left.value).to.equal(1);
+        expect(params[0].right.operator).to.equal('+');
+        expect(params[0].right.right.type).to.equal('Literal');
+        expect(params[0].right.right.value).to.equal(2);
+    });
```
Comment (mdevils) [inline test/lib/elements/types/AssignmentPattern.js]: You can also add `{x, y} = {x: 1, y: 2}` and `[x, y] = [1, 2]`.

**Comment by mdevils** [comment]
> Is there a site/docs/something that has coverage over all syntax we could use? Then we can get more coverage.

Not yet. It's quite problematic right now as we don't have a good source for ES syntax documentation (with both structure and examples). But I hope we can generate something like that using our classes and JSDoc.

File: test/lib/elements/types/AssignmentPattern.js (hunk 1 above)
Comment (hzoo) [inline test/lib/elements/types/AssignmentPattern.js]: You mean destructuring with defaults?

`let {x = 0, y = 3} = {x: 1, y: 2}`

Comment (mdevils) [inline test/lib/elements/types/AssignmentPattern.js]: No, this syntax is supported in function params if I'm not mistaken.

Comment (hzoo) [inline test/lib/elements/types/AssignmentPattern.js]: Ah I see makes sense

Comment (hzoo) [inline test/lib/elements/types/AssignmentPattern.js]: Updated with your suggestion

**Comment by hzoo** [comment]
weird issue with `nvm install 0.12` not working.

**Comment by mdevils** [comment]
Restarted travis job. Now it's OK.

**Comment by hzoo** [comment]
Should be good then. But yeah it looks like all our current tests are just against the ast and not whitespace?

**Comment by mdevils** [comment]
What do you mean by "just against the ast and not whitespace"?

**Comment by hzoo** [comment]
Like are we supposed to assert there is a certain amount of whitespace for each piece of code or is that not relevant?

**Comment by mdevils** [comment]
> I guess that part is covered by skipNonCode tests so never mind.

Yes, this method completely covers whitespace.

**Comment by mdevils** [comment]
We only check if that method is called when necessary (i.e. both test `x+y` and `x + y`).

**Pull request merged**